### PR TITLE
PCCS service containerization with non-interactive configuration

### PR DESCRIPTION
**Pull request by dpuzikox**
I hit a need to run pccs as a container but there's no option to configure pccs non-interactively.
This PR should improve this situation.

### CODE REVIEW

File: QuoteGeneration/pccs/container/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# build libPCKCertSelection library and copy to lib folder
+RUN make && \
+    mkdir -p ../../QuoteGeneration/pccs/lib && \
+    cp ./out/libPCKCertSelection.so ../../QuoteGeneration/pccs/lib/ && \
+    make clean
+
+# set PWD to PCCS dir
+WORKDIR /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/
+
```
Comment (haitaohuang) [inline QuoteGeneration/pccs/container/Dockerfile:41]: I suggest docker multi-stage build (https://docs.docker.com/develop/develop-images/multistage-build/) to minimize dependencies at runtime.

File: QuoteGeneration/pccs/container/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Dockerfile
+.dockerignore
+/config
```
Comment (haitaohuang) [inline QuoteGeneration/pccs/container/.dockerignore:3]: add EOF or empty line at the end

File: QuoteGeneration/pccs/container/Dockerfile (hunk 1 above)
Comment (dpuzikox) [reply QuoteGeneration/pccs/container/Dockerfile:38]: Fixed.

**Comment by haitaohuang** [comment]
Looks good to me

**Comment by lingyuj** [comment]
The patch looks fine. Will ask maintainer to merge it.

File: QuoteGeneration/pccs/container/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            zip
+
+# install node.js
+RUN curl -sL https://deb.nodesource.com/setup_14.x | bash -
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends nodejs
+
+RUN apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# clone DCAP repo
+RUN git clone https://github.com/intel/SGXDataCenterAttestationPrimitives.git -b ${DCAP_VERSION} && \
```
[the remark this answers is not in the document]
Comment (dpuzikox) [reply QuoteGeneration/pccs/container/Dockerfile:24]: I'm  choosing `--depth 1` to keep things simple. 
Just don't want to bother with conditions to select correct url to curl repo: /tag/ or /heads/  or /.
Fixing.

File: QuoteGeneration/pccs/container/config/default.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
+{
```
[the remark this answers is not in the document]
Comment (dpuzikox) [reply QuoteGeneration/pccs/container/config/default.json:1]: Good catch! Not a single reason to include this file.

[the remark this answers is not in the document]
Comment (dpuzikox) [reply QuoteGeneration/pccs/container/config/default.json:1]: Replied ^^^ . Will remove and fix documentation referring to existing `QuoteGeneration/pccs/config/default.json`

File: QuoteGeneration/pccs/container/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+mkdir -p ./config
+```
+Copy `<path_to_repo>/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/config/default.json`
+to this directory:
+```
+cp <path_to_repo>/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/config/default.json ./config/
+```
+Generate UserTokenHash:
+```
+echo "user_password" | sha512sum | tr -d '[:space:]-'
```
Comment (Nirusu) [inline QuoteGeneration/pccs/container/README.md:32]:
```suggestion
echo -n "user_password" | sha512sum | tr -d '[:space:]-'
```

You don't want to include newlines into your password...

File: QuoteGeneration/pccs/container/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+```
+and AdminTokenHash:
+```
+echo "admin_password" | sha512sum | tr -d '[:space:]-'
```
Comment (Nirusu) [inline QuoteGeneration/pccs/container/README.md:36]:
```suggestion
echo -n "admin_password" | sha512sum | tr -d '[:space:]-'
```
Same here

**Review by Nirusu — CHANGES_REQUESTED** [review]

File: QuoteGeneration/pccs/container/README.md (hunk 5 above)
Comment (dpuzikox) [reply QuoteGeneration/pccs/container/README.md]: Fixed.

File: QuoteGeneration/pccs/container/README.md (hunk 6 above)
Comment (dpuzikox) [reply QuoteGeneration/pccs/container/README.md]: Fixed.

**Review by dcmiddle — APPROVED** [review]

**Review by Nirusu — APPROVED** [review]

File: QuoteGeneration/pccs/container/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,61 @@
+FROM ubuntu:20.04 AS builder
+
+# DCAP version (github repo branch, tag or commit hash)
+ARG DCAP_VERSION=DCAP_1.10.3
```
[the remark this answers is not in the document]
Comment (dpuzikox) [reply QuoteGeneration/pccs/container/Dockerfile]: @jly77 -  Done.
@lzha101 - Sign-off-by filed added.

**Pull request merged**